src/System.Management.Automation/engine/Subsystem/FeedbackSubsystem/IFeedbackProvider.cs

```diff
@@ -0,0 +1,285 @@
+                var file = new FileInfo(path);
+                return file.Exists && file.UnixFileMode.HasFlag(UnixFileMode.OtherExecute);
+            }
+        }
+
+        /// <summary>
+        /// Gets feedback based on the given commandline and error record.
+        /// </summary>
+        public string? GetFeedback(string commandLine, ErrorRecord lastError, CancellationToken token)
+        {
+            if (Platform.IsWindows || lastError.FullyQualifiedErrorId != "CommandNotFoundException")
+            {
+                return null;
+            }
+
+            var target = (string)lastError.TargetObject;
+            if (target is null)
+            {
+                return null;
+            }
+
+            if (target.EndsWith(".ps1", StringComparison.OrdinalIgnoreCase))
+            {
+                return null;
+            }
+
+            string? cmd_not_found = GetUtilityPath();
+            if (cmd_not_found is not null)
+            {
+                var startInfo = new ProcessStartInfo(cmd_not_found);
+                startInfo.ArgumentList.Add(target);
+                startInfo.RedirectStandardError = true;
+                startInfo.RedirectStandardOutput = true;
+
+                using var process = Process.Start(startInfo);
+                var stderr = process?.StandardError.ReadToEnd().Trim();
+
+                // The feedback contains recommended actions only if the output has multiple lines of text.
+                if (stderr?.IndexOf('\n') > 0)
+                {
+                    _notFoundFeedback = stderr;
+
+                    var stdout = process?.StandardOutput.ReadToEnd().Trim();
+                    return string.IsNullOrEmpty(stdout) ? stderr : $"{stderr}\n{stdout}";
+                }
+            }
+
+            return null;
+        }
+
+        #endregion
+
+        #region ICommandPredictor
+
+        public bool CanAcceptFeedback(PredictionClient client, PredictorFeedbackKind feedback)
+        {
+            return feedback switch
+            {
+                PredictorFeedbackKind.CommandLineAccepted => true,
+                _ => false,
+            };
+        }
+
+        public SuggestionPackage GetSuggestion(PredictionClient client, PredictionContext context, CancellationToken cancellationToken)
+        {
+            if (_candidates is null && _notFoundFeedback is not null)
+            {
+                var text = _notFoundFeedback.AsSpan();
+                // Set to null to avoid potential race condition.
+                _notFoundFeedback = null;
+
+                // This loop searches for candidate results with almost no allocation.
+                while (true)
+                {
+                    // The line is a candidate if it starts with "sudo ", such as "sudo apt install python3".
+                    // 'sudo' is a command name that remains the same, so this check should work for all locales.
+                    bool isCandidate = text.StartsWith("sudo ");
```

Comment:
Is this intentionally Culture and case-sensetive comparison?

Reply:
Good catch. It's supposed to use `StringComparison.Ordinal`.